Sources/App/Models/SocialAccount/Social.swift

```diff
@@ -0,0 +1,94 @@
+import Vapor
+import FluentProvider
+import HTTP
+
+final class Social: Model {
+  let storage = Storage()
+  
+  var name: String
+  var appId: String
+  var secureKey: String
+  var serviceToken: String
+  
+  struct Keys {
+    static let id = "id"
+    static let name  = "name"
+    static let appId = "app_id"
+    static let secureKey = "secure_key"
+    static let serviceToken = "service_token"
+  }
+  
+  init(name: String, appId: String, secureKey: String, serviceToken: String) {
+    self.name = name
+    self.appId = appId
+    self.secureKey = secureKey
+    self.serviceToken = serviceToken
+  }
+  
+  init(row: Row) throws {
+    name = try row.get(Social.Keys.name)
+    appId = try row.get(Social.Keys.appId)
+    secureKey = try row.get(Social.Keys.secureKey)
+    serviceToken = try row.get(Social.Keys.serviceToken)
+    id = try row.get(Social.Keys.id)
+  }
+  
+  func makeRow() throws -> Row {
+    var row = Row()
+    try row.set(Social.Keys.id, id)
+    try row.set(Social.Keys.name, name)
+    try row.set(Social.Keys.appId, appId)
+    try row.set(Social.Keys.secureKey, secureKey)
+    try row.set(Social.Keys.serviceToken, serviceToken)
+    return row
+  }
+}
+
+// MARK: Relations
+
+extension Social {
+  var accounts: Children<Social, SocialAccount> {
```

Comment:
In general, we don't need accounts for special social network. On the contrary, we need social network data to the social account model. So, you can create relationship from social account to the social network.